konan_sdk/konan_service/services.py

```diff
@@ -0,0 +1,53 @@
+from typing import Type
+
+from fastapi import FastAPI, Response
+from fastapi_utils.cbv import cbv
+
+from konan_sdk.konan_service.routers import KonanServiceRouter
+
+
+class KonanService():
+    """Class that implements a Konan webservice
+    """
+    def __init__(
+        self,
+        predict_request_class: Type,
+        predict_response_class: Type,
+        model_class: Type,
+        *model_args, **model_kwargs,
+    ) -> None:
+        """Initializes a konan service
+
+        Args:
+            predict_request_class (Type): Type of a prediction request.
+                Should be a class that inherits from KonanServiceBasePredictionRequest
+            predict_response_class (Type): Type of a prediction response.
+                Should be a class that inherits from KonanServiceBasePredictionResponse
+            model_class (Type): Type of the model that does the prediction.
+                Should be a class that inherits from KonanServiceBaseModel.
+                Must implement the predict() method
+        """
+        self.model = model_class(*model_args, **model_kwargs)
```

Comment:
Minor point, wouldn't it be simpler to just let the user pass an instantiated model instance?  Unless we plan to instantiate the multiple of times

Reply:
Yeah, we won't be instantiating it multiple times, especially since scaling the application itself will be done by us automatically. 
However, I would rather keep it like this for now to have all instantiations self-contained inside the SDK, and present to the end-user a consistent API of them having to just _define_ their ML logic, and all other steps being automatically handled by us.